Formula/sapling.rb

```diff
@@ -0,0 +1,33 @@
+class Sapling < Formula
+  desc "Source control client"
+  homepage "https://sapling-scm.com"
+  url "https://github.com/facebook/sapling/archive/refs/tags/0.1.20221118-210929-cfbb68aa.tar.gz"
```

Comment:
The release tag seems a bit off, I think upstream should document it?

Reply:
Would something like

```
  url "https://github.com/facebook/sapling/archive/0.1.20221118-210929-cfbb68aa.tar.gz"
```

 work any better?

Reply:
nope, I am mostly referring to this `0.1.20221118-210929-cfbb68aa`, which made you need a `0.1.20221118` to specify the version

Reply:
I manually specified `version "0.1.20221118-210929-cfbb68aa"` since Homebrew wasn't able to pick up the last part in the version. For instance, without this, the `0.1.20221118-210929-12345ab` tag would be shortened to `0.1.20221118-210929-12345`. I'm still not sure why Homebrew was chopping off letters in the end of the version.

Our versioning scheme (`MAJOR.MINOR.%Y%m%d-%H%M%S-HASH`) seems to comply with [Semantic Versioning 2.0.0](https://semver.org/), which AFAIK is widely accepted.

Reply:
Surprisingly I only learned this semver 2.0 today, I was only family with `major.minor.patch` scheme 😆

Reply:
> I'm still not sure why Homebrew was chopping off letters in the end of the version.

Let me see if I can find the reference in the `brew` code

Reply:
This would be the method ([`_parse`](https://github.com/Homebrew/brew/blob/9c88c39baead6bbc40b12ab1eae9b63bcb5a8042/Library/Homebrew/version.rb#L369)) that you are looking for to parse out the version from url

Reply:
maybe file a PR for this? 

https://github.com/Homebrew/brew/blob/9c88c39baead6bbc40b12ab1eae9b63bcb5a8042/Library/Homebrew/version.rb#L395

Reply:
Thanks for the suggestion. Looking at some examples in that file, we decided to rather move to the `MAJOR.MINOR.%Y%m%d-%H%M%S-rHASH` instead.